x-pack/plugins/fleet/server/types/extensions.ts

```diff
@@ -7,16 +7,16 @@
 
 import type { KibanaRequest, RequestHandlerContext } from 'kibana/server';
 
+import type { DeepReadonly } from 'utility-types';
+
 import type {
   DeletePackagePoliciesResponse,
   NewPackagePolicy,
   UpdatePackagePolicy,
 } from '../../common';
 
 export type PostPackagePolicyDeleteCallback = (
-  deletedPackagePolicies: DeletePackagePoliciesResponse,
-  context: RequestHandlerContext,
-  request: KibanaRequest
+  deletedPackagePolicies: DeepReadonly<DeletePackagePoliciesResponse>
```

Comment:
Nice 👏   I should remember to use `utility-types`. Although, I'm quite sure `Readonly<DeletePackagePoliciesResponse>` should work the same way here. 🤔

Reply:
`utility-types` has some nice utilities that you should look at anytime you are trying to do something that "feels" like TS should support, but does not.

Here, I want everything (including deep structures) to be readonly. the TS `Readonly<>` generic will only make the shallow keys of the type provided readonly, but not deeper structures.